---
title:  "MODEL_COMPARISON_FIGURE1"
author: "Matthew J Shannon"
date:   "18/01/2024"
output: html_document
---

# Load in all dependencies

```{r, include = FALSE}
devtools::install_github ('chris-mcginnis-ucsf/DoubletFinder')
devtools::install_github ('satijalab/seurat-wrappers')
remotes::install_github  ("mojaveazure/seurat-disk")
devtools::install_github ("velocyto-team/velocyto.R")
BiocManager::install     ("pcaMethods")
BiocManager::install     ('MAST')
BiocManager::install     ('org.Hs.eg.db')
BiocManager::install     (version = "3.12")
BiocManager::install     ('Seurat')
BiocManager::install     ('readxl')
BiocManager::install     ('modes')
BiocManager::install     ('pheatmap')
BiocManager::install     ('limma')
BiocManager::install     ('clustree')
BiocManager::install     ('clusterProfiler')
BiocManager::install     ('EnhancedVolcano')
install.packages         ('fields')
install.packages         ('plotly')
install.packages         ("VennDiagram")
library                  (Seurat)
library                  (readr)
library                  (tidyverse)
library                  (Matrix)
library                  (dplyr)
library                  (ggplot2)
library                  (cowplot)
library                  (tibble)
library                  (readxl)
library                  (sctransform)
library                  (fields)
library                  (KernSmooth)
library                  (ROCR)
library                  (parallel)
library                  (reshape2)
library                  (pheatmap)
library                  (DoubletFinder)
library                  (limma)
library                  (SeuratWrappers)
library                  (SeuratDisk)
library                  (plotly)
library                  (clustree)
library                  (velocyto.R)
library                  (MAST)
library                  (EnhancedVolcano)
library                  (clusterProfiler)
library                  (AnnotationDbi)
library                  (org.Hs.eg.db)
library                  (VennDiagram)
library                  (RColorBrewer)
library                  (xlsx)
gc                       ()
```

# Load the integrated trophoblast Seurat object

```{r}
# Integrated (all models) trophoblast object
load         (file = "~/Desktop/FINAL_Trophoblast_Data.Rdata")
DefaultAssay (Trophoblasts) <- "integrated"
```

# Figure 1

## Integrated trophoblast data analyses

```{r, Table S5A Integrated data}
table (Idents (Trophoblasts))
```

```{r, Figure 1D Integrated UMAP}
UMAPPlot (Trophoblasts,
          label   = TRUE,
          pt.size = 3,
          cols    = c('CTB 1'  = '#E31A1C',
                      'CTB 2'  = '#999999',
                      'CTB 3'  = '#225EA8',
                      'CTB 4'  = '#984EA3',
                      'TSC'    = '#A65628',
                      'cCTB 1' = '#66C2A4',
                      'cCTB 2' = '#FFD92F',
                      'cCTB 3' = '#FF7F00',
                      'EVT'    = '#006D2C',
                      'SCTp 1' = '#F781BF',
                      'SCTp 2' = '#800026')) + NoLegend ()
```

## Integrated trophoblast (un-biased) gene markers

```{r, Table S3A analysis}
DefaultAssay (Trophoblasts) <- "RNA"

Trophoblasts.markers <- FindAllMarkers (object   = Trophoblasts,
                                        only.pos = TRUE,
                                        min.pct  = 0.25, 
                                        test.use = "MAST")

DefaultAssay (Trophoblasts) <- "integrated"

Trophoblasts.markers %>% group_by (cluster) %>% top_n (20, avg_log2FC)
```

```{r, Table S3A export}
write.xlsx (Trophoblasts.markers, file = "Integrated_genemarkers.xlsx")
```

### MPI (in vivo) data

```{r, Table S5A MFI-specific data}
MPI <- subset (Trophoblasts, subset = Model == "MPI")
table (Idents (MPI))
```

```{r, Figure 1D MFI-specific UMAP}
UMAPPlot (MPI,
          label   = TRUE,
          pt.size = 3,
          cols    = c('CTB 1'  = '#E31A1C',
                      'CTB 2'  = '#999999',
                      'CTB 3'  = '#225EA8',
                      'CTB 4'  = '#984EA3',
                      'TSC'    = '#A65628',
                      'cCTB 1' = '#66C2A4',
                      'cCTB 2' = '#FFD92F',
                      'cCTB 3' = '#FF7F00',
                      'EVT'    = '#006D2C',
                      'SCTp 1' = '#F781BF',
                      'SCTp 2' = '#800026')) + NoLegend ()
```

### TBpOrg (in vitro) data

```{r, Table S5A PD-TOrg-specific data}
TBpOrg <- subset (Trophoblasts, subset = Model == "TBpOrg")
table (Idents (TBpOrg))
```

```{r, Figure 1D PD-TOrg-specific UMAP}
UMAPPlot (TBpOrg,
          label   = TRUE,
          pt.size = 3,
          cols    = c('CTB 1'  = '#E31A1C',
                      'CTB 2'  = '#999999',
                      'CTB 3'  = '#225EA8',
                      'CTB 4'  = '#984EA3',
                      'TSC'    = '#A65628',
                      'cCTB 1' = '#66C2A4',
                      'cCTB 2' = '#FFD92F',
                      'cCTB 3' = '#FF7F00',
                      'EVT'    = '#006D2C',
                      'SCTp 1' = '#F781BF',
                      'SCTp 2' = '#800026')) + NoLegend ()
```

### TBsOrg (in vitro) data

```{r, Table S5A hTSC-TOrg-specific data}
TBsOrg <- subset (Trophoblasts, subset = Model == "TBsOrg")
table (Idents (TBsOrg))
```

```{r, Figure 1D hTSC-TOrg-specific UMAP}
UMAPPlot (TBsOrg,
          label   = TRUE,
          pt.size = 3,
          cols    = c('CTB 1'  = '#E31A1C',
                      'CTB 2'  = '#999999',
                      'CTB 3'  = '#225EA8',
                      'CTB 4'  = '#984EA3',
                      'TSC'    = '#A65628',
                      'cCTB 1' = '#66C2A4',
                      'cCTB 2' = '#FFD92F',
                      'cCTB 3' = '#FF7F00',
                      'EVT'    = '#006D2C',
                      'SCTp 1' = '#F781BF',
                      'SCTp 2' = '#800026')) + NoLegend ()
```

## Holistic model transcriptome-comparison

```{r, trophoblast model-specific Pearson Correlation Coefficient analysis, data not plotted in paper}
Idents (Trophoblasts) <- Trophoblasts@meta.data$Model
av.exp                <- AverageExpression (Trophoblasts)$RNA
cor.exp               <- as.data.frame     (cor (av.exp))
cor.exp$x             <- rownames          (cor.exp)
cor.df                <- tidyr::gather     (data = cor.exp, y, correlation, c('MPI', 'TBpOrg', 'TBsOrg'))
cor.df

ggplot               (cor.df, aes (x, y, fill = correlation)) +
geom_tile            () +
scale_fill_gradientn (limits = c(0, 1), colours = colorRampPalette (rev (brewer.pal (n = 7, name = "RdYlBu"))) (100))
```

## Integrated cell composition comparison

```{r, Integrated trophoblast state binning}
# Label integrated trophoblast states cell type
             new.cluster.ids <- c('CTB 1'  = "CTB",
                                  'CTB 2'  = "CTB",
                                  'CTB 3'  = "CTB",
                                  'CTB 4'  = "CTB",
                                  'cCTB 1' = "PTB",
                                  'cCTB 2' = "cCTB",
                                  'cCTB 3' = "cCTB",
                                  'cCTB 4' = "cCTB",
                                  'EVT'    = "EVT",
                                  'SCTp 1' = "SCTp",
                                  'SCTp 2' = "SCTp")
# Apply binned trophoblast labels to integrated Seurat object
names (new.cluster.ids)      <- levels       (Trophoblasts)
Trophoblasts$seurat_clusters <- Idents       (Trophoblasts)
Trophoblasts                 <- RenameIdents (Trophoblasts, new.cluster.ids)
```

```{r, Figure 1F}
ggplot             (Trophoblasts@meta.data, aes (x = c(Trophoblasts$Model), fill = Trophoblasts$celltype)) +
theme_classic2     () +
geom_bar           (position     = "fill") +
scale_y_continuous (labels       = scales::percent) +
theme              (text         = element_text (size   = 35)) +
theme              (axis.text    = element_text (size   = 35)) +
theme              (axis.title.x = element_text (face   = "bold")) +
theme              (axis.title.y = element_text (face   = "bold")) +
theme              (legend.text  = element_text (size   = 15)) +
theme              (axis.text.y  = element_text (angle  = 90)) +
theme              (axis.text.y  = element_text (hjust  = 0.5)) +
theme              (axis.text.x  = element_text (colour = "black")) +
theme              (axis.text.y  = element_text (colour = "black")) +
xlab               ("Trophoblast Model") +
ylab               ("Cell Proportion")
```

## Integrated trophoblast gene expression comparison

```{r, Figure 1G, analysis}
Trophoblasts$celltype.model  <- paste       (Idents (Trophoblasts), Trophoblasts$Model, sep = "_")
Trophoblasts$celltype        <- Idents      (Trophoblasts)

       Idents (Trophoblasts) <- "celltype.model"

# Table S5A
table (Idents (Trophoblasts))
```

```{r, Figure 1G}
DefaultAssay (Trophoblasts) <- "RNA"
Idents       (Trophoblasts) <- "celltype.model"

my_levels <- c("CTB 4_MPI",  "CTB 4_TBpOrg",  "CTB 4_TBsOrg",
               "CTB 3_MPI",  "CTB 3_TBpOrg",  "CTB 3_TBsOrg",
               "CTB 2_MPI",  "CTB 2_TBpOrg",  "CTB 2_TBsOrg",
               "CTB 1_MPI",  "CTB 1_TBpOrg",  "CTB 1_TBsOrg",
                 "TSC_MPI",    "TSC_TBpOrg",    "TSC_TBsOrg",
              "cCTB 1_MPI", "cCTB 1_TBpOrg", "cCTB 1_TBsOrg",
              "cCTB 2_MPI", "cCTB 2_TBpOrg", "cCTB 2_TBsOrg",
              "cCTB 3_MPI", "cCTB 3_TBpOrg", "cCTB 3_TBsOrg",
                 "EVT_MPI",    "EVT_TBpOrg",    "EVT_TBsOrg",
              "SCTp 1_MPI", "SCTp 1_TBpOrg", "SCTp 1_TBsOrg",
              "SCTp 2_MPI", "SCTp 2_TBpOrg", "SCTp 2_TBsOrg")

Trophoblasts@active.ident <- factor (Trophoblasts@active.ident, levels = my_levels)

DotPlot (Trophoblasts,
         features  = c("CGB",
                       "ERVFRD-1",
                       "SDC1",
                       "HLA-G",
                       "ITGA5",
                       "ITGA1",
                       "NOTCH2",
                       "ITGA2",
                       "NOTCH1",
                       "EPCAM",
                       "TEAD4",
                       "TP63",
                       "BCAM",
                       "ELF5",
                       "TFAP2A",
                       "TFAP2C",
                       "GATA2",
                       "GATA3",
                       "MKI67",
                       "CCNA2"),
         assay     = "RNA",
         cols      = c("lightgrey", "red"),
         col.min   = 0,
         col.max   = 1,
         dot.min   = 0,
         dot.scale = 15)
```
